3.2 Backport of Fix: Get proper mFeature in relation reference widget

signedav opened the pull request:
The issue was, that the embedded attribute form of the relation reference didn't got updated in case there was the first entry in the combo box selected (because feature was not valid).
With this fix we update the embedded form after the initialization.

IMHO this should have been needed before and not only for the first entry in the combo box. But because the combo box gets an update in case it's not the first entry, the signal is fired to update the combo box and so the embedded form. In the older version it seems the combo box got rearranged even when it's the first entry, and so it worked back then.

Anyway the embedded form is updated (and the related feature highlighted) now in any case after the initialization.

Fix #19342

Backport from #7398

signedav committed:
Get mFeature before updating attributeform

The issue was, that the attributeform didn't got updated in case there was the first entry in the combobox selected.
If there was not the first combobox entry selected, there has been a setindex after the initialization and the attributeform got updated.
But the first entry needs this call.

Fix #19342

(cherry-picked from 4c2e5ff4f582aae8a533d962e3c18c034420d8ac)

signedav committed:
highlight on initialization

(cherry-picked from 9ff71a0a9f764396ca9bba82d0c285d87ed44601)

signedav committed:
emit currentIndexChanged onDataChanged

Because this effects that the data related to the combobox (emedded forms) got updated. they update because of an index change, but needed to be updated even when the index does not change (but the data). This goes over the same canal here.

(cherry-picked from 56fe370e851f3fdeb5515f6dc2f8cf794ed47155)

signedav committed:
initialization of embeddet form only once

(cherry-picked from 56fe370e851f3fdeb5515f6dc2f8cf794ed47155)

signedav committed:
fix spelling

(cherry-picked from 56fe370e851f3fdeb5515f6dc2f8cf794ed47155)

signedav commented:
Since Matthias is not available and this should be on the 3.2.1, could someone else review this? Could you suggest someone @nyalldawson ? Thanks.

nyalldawson commented:
@signedav I'm not personally familiar enough with this area of the code to make a good call here. But I think the backport should wait until #7398 is reviewed and merged -- I'm not comfortable with merging a backport before the corresponding PR is reviewed.

Maybe @3nids can review #7398 in Matthias' absence?

signedav committed:
use initialized for complete init() function

not only for the feature part

(cherry-picked from 4739463426962576417bfbd72b1336aed67221df)
